src/csharp/Microsoft.Spark/Sql/SparkSession.cs

```diff
@@ -61,10 +61,39 @@ internal SparkSession(JvmObjectReference jvmObject)
         /// <returns>Builder object</returns>
         public static Builder Builder() => new Builder();
 
-        /// Note that *ActiveSession() APIs are not exposed because these APIs work with a
-        /// thread-local variable, which stores the session variable. Since the Netty server
-        /// that handles the requests is multi-threaded, any thread can invoke these APIs,
-        /// resulting in unexpected behaviors if different threads are used.
+        /// <summary>
+        /// Changes the SparkSession that will be returned in this thread and its children when
+        /// SparkSession.GetOrCreate() is called. This can be used to ensure that a given thread
+        /// receives a SparkSession with an isolated session, instead of the global (first created)
+        /// context.
+        /// </summary>
+        /// <param name="session"></param>
+        public static void SetActiveSession(SparkSession session) =>
+            session._jvmObject.Jvm.CallStaticJavaMethod(
+                s_sparkSessionClassName, "setActiveSession", session);
+
+        /// <summary>
+        /// Clears the active SparkSession for current thread. Subsequent calls to getOrCreate will
+        /// return the first created context instead of a thread-local override.
```

Comment:
```suggestion
        /// Clears the active SparkSession for current thread. Subsequent calls to
        /// <see cref="Builder.GetOrCreate"/> will return the first created context
        /// instead of a thread-local override.
```